### PR TITLE
nx-list of clickable links - RSC-525

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-525

I realized the need for this while reviewing https://github.com/sonatype/insight-brain/pull/6201

### CODE REVIEW

File: lib/src/base-styles/_nx-lists.scss [hunk 1]
```diff
@@ -49,32 +49,42 @@
...
   display: block;
-  -ms-grid-row: 1;
-  -ms-grid-column: 2;
   grid-area: list-text;
   margin: 0;
   padding-right: $nx-spacing-l;
+
+  &.nx-truncate-ellipsis {
+    // Something about the grid layout gets confused due to overflow: hidden on this element
+    //  and adds more vertical space to the grid, in the second row.
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-lists.scss:78]:
```suggestion
    //  and adds more vertical space to the second row of the grid.
```

File: lib/src/base-styles/_nx-lists.scss [hunk 2]
```diff
@@ -49,32 +49,42 @@
...
+    // Experimentally, setting an explicit height here fixes that.
```
Comment (glenhunter) [inline lib/src/base-styles/_nx-lists.scss:79]:
```suggestion
    // Setting an explicit height here fixes that.
```

File: gallery/src/styles/NxList/NxListExamples.tsx [hunk 3]
```diff
@@ -38,6 +40,14 @@ const NxListsExamples = () =>
       An <code className="nx-code">nx-list</code> demonstrating clickable, selection, and disabled styles.
     </GalleryExampleTile>
 
+    <GalleryExampleTile title="NX Clickable List Links Example"
+                        id="nx-list-clickable-links-example"
+                        liveExample={NxListClickableLinksExample}
+                        codeExamples={NxListClickableLinksCode}>
+      An <code className="nx-code">nx-list</code> demonstrating clickable and selection styles where each
+      list item is effectively a link.
```
Comment (glenhunter) [inline gallery/src/styles/NxList/NxListExamples.tsx:48]:
```suggestion
      An <code className="nx-code">nx-list</code> demonstrating clickable and selection styles where the
      clickable aspects of the list item are defined using an `&lta&gt; tag.
```

**Review by glenhunter — APPROVED** [review]
Approved with minor text tweaks

File: lib/src/base-styles/_nx-lists.scss (hunk 2 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-lists.scss:79]: > // Setting an explicit height here fixes that.

I disagree with this one.  Pointing out that we don't know why this worked is valuable.

File: lib/src/base-styles/_nx-lists.scss (hunk 1 above)
Comment (rpokorny) [reply lib/src/base-styles/_nx-lists.scss]: > and adds more vertical space to the second row of the grid.

ecf755aa

File: gallery/src/styles/NxList/NxListExamples.tsx (hunk 3 above)
Comment (rpokorny) [reply gallery/src/styles/NxList/NxListExamples.tsx]: > clickable aspects of the list item are defined using an `&lta&gt; tag.

ecf755aa

**Pull request merged**